apis/management.cattle.io/v3/pipeline_types.go

```diff
@@ -78,14 +82,15 @@ type GithubClusterConfig struct {
 }
 
 type PipelineStatus struct {
-	PipelineState   string `json:"pipelineState,omitempty" norman:"required,options=active|inactive,default=active"`
-	NextRun         int    `json:"nextRun" yaml:"nextRun,omitempty" norman:"default=1,min=1"`
-	LastExecutionID string `json:"lastExecutionId,omitempty" yaml:"lastExecutionId,omitempty"`
-	LastRunState    string `json:"lastRunState,omitempty" yaml:"lastRunState,omitempty"`
-	LastStarted     string `json:"lastStarted,omitempty" yaml:"lastStarted,omitempty"`
-	NextStart       string `json:"nextStart,omitempty" yaml:"nextStart,omitempty"`
-	WebHookID       string `json:"webhookId,omitempty" yaml:"webhookId,omitempty"`
-	Token           string `json:"token,omitempty" yaml:"token,omitempty" norman:"writeOnly,noupdate"`
+	PipelineState        string                `json:"pipelineState,omitempty" norman:"required,options=active|inactive,default=active"`
+	NextRun              int                   `json:"nextRun" yaml:"nextRun,omitempty" norman:"default=1,min=1"`
+	LastExecutionID      string                `json:"lastExecutionId,omitempty" yaml:"lastExecutionId,omitempty"`
+	LastRunState         string                `json:"lastRunState,omitempty" yaml:"lastRunState,omitempty"`
+	LastStarted          string                `json:"lastStarted,omitempty" yaml:"lastStarted,omitempty"`
+	NextStart            string                `json:"nextStart,omitempty" yaml:"nextStart,omitempty"`
+	WebHookID            string                `json:"webhookId,omitempty" yaml:"webhookId,omitempty"`
+	Token                string                `json:"token,omitempty" yaml:"token,omitempty" norman:"writeOnly,noupdate"`
+	SourceCodeCredential *SourceCodeCredential `json:"sourceCodeCredential,omitempty" yaml:"sourceCodeCredential,omitempty"`
```

Comment:
Why not use a id here?  and set the field `norman:"type:reference[sourceCodeCredential]"`

Reply:
As stated in the PR description:
>Because Pipelines are in project-scoped. When user A sees user B's pipelines we need some info of B's Github account to show(like avatar, Github username, etc.), so we store a sourcecodecredential reference in pipelinestatus with sensitive data erased.

Reply:
ok, got it.  If the sourceCodeCredential is changed, will update the status here?

Reply:
It should, but we don't expect it to be changeable by users.